### PR TITLE
The first attempt to add automatedtests

### PR DESCRIPTION
**Pull request by AnnaLangner**
I know I have something wrong because I don't get any error. Should I split the `test_search_in_wikipedia_org` function into several smaller ones?

### CODE REVIEW

File: script/test_2.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+
+
+def setUp(self):
```
Comment (klangner) [inline script/test_2.py]: Only class gets as a first argument `self`.
Here it is not necessary. Usually it is not worth to create function for single line of code.
But if you really want one then in this case it is better to just return initialized driver then use classes and class members.

**Review by klangner — CHANGES_REQUESTED** [review]
Does it work for you?
It doesn't look like :-)

File: script/test_2.py (hunk 1 above)
Comment (AnnaLangner) [reply script/test_2.py]: I think it doesn't work, but I don't know why because I don't have any error :(

Comment (klangner) [reply script/test_2.py]: You simply don't call any functions here. And functions are not executed by them self.

Comment (AnnaLangner) [reply script/test_2.py]: facepalm xD

Comment (AnnaLangner) [reply script/test_2.py]: I have this error again: `selenium.common.exceptions.WebDriverException: Message: 'chromedriver' executable needs to be in PATH. Please see https://sites.google.com/a/chromium.org/chromedriver/home` :(

Comment (klangner) [reply script/test_2.py]: How do you run this script?
Also push the current code, so I can look at it.

Comment (AnnaLangner) [reply script/test_2.py]: Push is done 

I run this code by comand : `python test_2.py`

Comment (AnnaLangner) [reply script/test_2.py]: When I moved "chromeedriver" to the "script" folder, it was all right

Comment (klangner) [reply script/test_2.py]: Ok. But how do you run this script. Please provide the full command line.

Comment (AnnaLangner) [reply script/test_2.py]:
```
annal@LAPTOP-MN5V2CJL MINGW64 /d/Programowanie/pysandbox/script (selenium2)
$ python test_2.py
```

Comment (klangner) [reply script/test_2.py]: that's the problem. Please notice that there is no folder bin inside script folder.
You need to run it from top level project folder with the command
```sh
python script/test_2.py
```

Comment (AnnaLangner) [reply script/test_2.py]: It work now

File: script/test_2.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import re
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+  
+
+def test_search_in_wikipedia_org():
```
Comment (klangner) [inline script/test_2.py]: You should call this function `main`.

**Review by klangner — CHANGES_REQUESTED** [review]
This doesn't work.
Look at the console for errors.

File: script/test_2.py (hunk 2 above)
Comment (AnnaLangner) [reply script/test_2.py]: oki

**Comment by AnnaLangner** [comment]
I don't know how to do it `Verify that the page doesn't contain phrase: "no needed"`. By xpath?

File: script/test_2.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+def main():
+  driver = webdriver.Chrome(executable_path="bin/chromedriver")
+  driver.get("https://en.wikipedia.org/wiki/Main_Page")
+  search_input = driver.find_element_by_id("searchInput")
+  search_input.clear()
+  search_input.send_keys("Unit Testing")
+  search_input.submit()
+  if "Unit Testing" in driver.current_url:
+    print("We are on a good site")
```
Comment (klangner) [inline script/test_2.py]: this is not needed. It was ok before

File: script/test_2.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  searched_link = driver.find_element_by_link_text("halting problem")
+  searched_link.click()
```
Comment (klangner) [inline script/test_2.py]: It is ok now

File: script/test_2.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  phrase_1 = "Turing machine"
+  phrase_2 = "no needed"
+  src = driver.page_source
+  found_phrase_1 = re.search(phrase_1, src)
```
Comment (klangner) [inline script/test_2.py]: Do not use re.search here. The phrase_1 is not a regex.
How else can you find something in the string?

**Review by klangner — CHANGES_REQUESTED** [review]
BTW If you want to add some general comments or have question, please use the issue for it.
It is hard to follow discussion in code review.

File: script/test_2.py (hunk 3 above)
Comment (AnnaLangner) [reply script/test_2.py]: I don't know how to make it work. When I wrote this:
```
 assert "Unit Testing" in driver.title
AssertionError
```
or I import unittest:
```
assertIn("Unit Testing", driver.title)
NameError: name 'assertIn' is not defined
```
it works only with if statement :(

Comment (klangner) [reply script/test_2.py]:
```python
assert "Unit Testing" in driver.title
```
Should be ok, what error do you get?

You have the same line in test_1.py and it works.
Maybe the problem is that the text is not in the title?

Comment (AnnaLangner) [reply script/test_2.py]: it  is in the title: `<title> Unit testing - Wikipedia </title>`
And that's why I don't understand it.
I only have "AssertionError"

Comment (klangner) [reply script/test_2.py]: Notice that it is written in different case :-)
If you don't care about letter cases than convert it to lower case first.

Comment (AnnaLangner) [reply script/test_2.py]: in fact, I didn't pay attention to it. I'm sorry

Comment (klangner) [reply script/test_2.py]: Nah. Not a problem. This can happen to anyone. :-)

File: script/test_2.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  driver.get("https://en.wikipedia.org/wiki/Main_Page")
+  search_input = driver.find_element_by_id("searchInput")
+  search_input.clear()
+  search_input.send_keys("Unit testing")
+  search_input.submit()
+  assert "Unit testing" in driver.title
+  searched_link = driver.find_element_by_link_text("halting problem")
+  searched_link.click()
+  src = driver.page_source
+  driver.find_elements_by_xpath("//*[contains(text(), 'Turing machine')]")
```
Comment (klangner) [inline script/test_2.py]: Check what this function returns `find_elements_by_xpath()`
And then in the assert use this information to check if the element was found or not.

File: script/test_2.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  search_input.clear()
+  search_input.send_keys("Unit testing")
+  search_input.submit()
+  assert "Unit testing" in driver.title
+  searched_link = driver.find_element_by_link_text("halting problem")
+  searched_link.click()
+  src = driver.page_source
+  driver.find_elements_by_xpath("//*[contains(text(), 'Turing machine')]")  
+  assert "No result found." not in src
+  driver.find_elements_by_xpath("//*[contains(text(), 'no needed')]")
```
Comment (klangner) [inline script/test_2.py]: Same here.

File: script/test_2.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  search_input = driver.find_element_by_id("searchInput")
+  search_input.clear()
+  search_input.send_keys("Unit testing")
+  search_input.submit()
+  assert "Unit testing" in driver.title
+  searched_link = driver.find_element_by_link_text("halting problem")
+  searched_link.click()
+  src = driver.page_source
+  driver.find_elements_by_xpath("//*[contains(text(), 'Turing machine')]")  
+  assert "No result found." not in src
```
Comment (klangner) [inline script/test_2.py]: Replace this with assert based on return value from the above function.

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/test_2.py [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  driver = webdriver.Chrome(executable_path="bin/chromedriver")
+  driver.get("https://en.wikipedia.org/wiki/Main_Page")
+  search_input = driver.find_element_by_id("searchInput")
+  search_input.clear()
+  search_input.send_keys("Unit testing")
+  search_input.submit()
+  assert "Unit testing" in driver.title
+  searched_link = driver.find_element_by_link_text("halting problem")
+  searched_link.click()
+  src = driver.page_source
```
Comment (klangner) [inline script/test_2.py]: This not not necessary. You don't need page_source here since you are using xpath.

**Review by klangner — CHANGES_REQUESTED** [review]

**Comment by AnnaLangner** [comment]
It does not work

File: script/test_2.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  driver.get("https://en.wikipedia.org/wiki/Main_Page")
+  search_input = driver.find_element_by_id("searchInput")
+  search_input.clear()
+  search_input.send_keys("Unit testing")
+  search_input.submit()
+  assert "Unit testing" in driver.title
+  searched_link = driver.find_element_by_link_text("halting problem")
+  driver.execute_script("arguments[0].click();", searched_link)
+  phrase_1 = driver.find_elements_by_xpath("//*[text()='Turing machine']")
+  print(phrase_1)
```
Comment (klangner) [inline script/test_2.py]: If it is working, then you can remove those print() functions

File: script/test_2.py [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+  assert len(phrase_1) > 0
+  phrase_2 = driver.find_elements_by_xpath("//*[text()='no needed']") 
+  print(phrase_2)
+  assert len(phrase_2) <= 0
```
Comment (klangner) [inline script/test_2.py]: it is better to be more precise here and write it as len(phrase_2) == 0

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/test_2.py (hunk 11 above)
Comment (AnnaLangner) [reply script/test_2.py]: oki :)

File: script/test_2.py (hunk 10 above)
Comment (AnnaLangner) [reply script/test_2.py]: I forgot about them

**Review by klangner — APPROVED** [review]
Well done!
